src/learning/gmm.rs

```diff
@@ -70,17 +70,25 @@ pub struct GaussianMixtureModel {
 impl UnSupModel<Matrix<f64>, Matrix<f64>> for GaussianMixtureModel {
     /// Train the model using inputs.
     fn train(&mut self, inputs: &Matrix<f64>) -> LearningResult<()> {
+        use rulinalg::matrix::Axes;
+
         // Initialization:
         let k = self.comp_count;
 
-        // TODO: Compute sample covariance of the data.
-        // https://en.wikipedia.org/wiki/Sample_mean_and_covariance#Sample_covariance
-        let mut cov_vec = Vec::with_capacity(k);
-        for _ in 0..k {
-            cov_vec.push(Matrix::identity(inputs.cols()));
+        let means = inputs.mean(Axes::Col);
+
+        let mut cov_mat = Matrix::zeros(inputs.cols(), inputs.cols());
+        let reg_value = 1f64 / (inputs.rows() - 1) as f64;
+
+        for j in 0..inputs.cols() {
+            for k in 0..inputs.cols() {
+                cov_mat[[j, k]] = (0..inputs.rows()).map(|i| {
```

Comment:
We can _safely_ use `get_unchecked(_mut)` here and below as we know we are within the dimensions.

Though I wonder if it would be better to construct the matrix by pushing to a `Vec` instead of zero-initialization (we access the data sequentially anyway).